### PR TITLE
Clone symint on set_sizes_and_strides

### PR DESCRIPTION
**Pull request by wconstab**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -109,6 +109,10 @@ def __init__(self, expr, shape_env, constant=None):
     def wrap(self, num):
         return PySymInt(sympy.Integer(num), self.shape_env, constant=num)
 
+    def clone(self):
+        import copy
+        return PySymInt(copy.deepcopy(self.expr), self.shape_env, constant=self.constant)
```
Comment (ezyang) [inline torch/fx/experimental/symbolic_shapes.py:114]: You don't need to clone the sympy expr

File: c10/core/TensorImpl.cpp [hunk 2]
```diff
@@ -912,6 +912,13 @@ bool _compute_contiguous(const ExtraMeta& extra_meta) {
   return _compute_contiguous(extra_meta, order);
 }
 
+void clone_symvec(SymIntArrayRef src,  SymDimVector& dst) {
+  dst.resize(src.size());
+  for (size_t i = 0; i < src.size(); i++){
+    dst[i] = src[i].toSymIntNodeImplUnowned()->clone()->toSymInt();
```
Comment (ezyang) [inline c10/core/TensorImpl.cpp:918]: It doesn't matter, but append is marginally better as it saves you from default initializing dest to 0, and then reupdating it in the loop.

File: c10/core/TensorImpl.cpp [hunk 3]
```diff
@@ -939,14 +946,20 @@ void TensorImpl::set_sizes_and_strides(
       extra_meta_->storage_offset_ = storage_offset_;
     }
   }
-  extra_meta_->sizes_ = sizes;
+  clone_symvec(sizes, extra_meta_->sizes_);
   extra_meta_->strides_ = strides;
+  // this segfaults
+  // clone_symvec(strides, extra_meta_->strides_);
```
Comment (ezyang) [inline c10/core/TensorImpl.cpp:952]: Hopefully these don't segfault anymore

Comment (wconstab) [reply c10/core/TensorImpl.cpp]: yea, the segfault was mainly coming from cases where the symint wasn't symbolic and i was trying to force a symbolic pointer out of it.  but the resize thing wasn't good either.  in any case it seems stable now

**Pull request merged**